### ISSUE TITLE
Windows Defender flagged Emora.Setup.exe as Trojan:Win32/Phonzy.C!ml

### ISSUE THREAD

**Issue by alistairwo** [Prescoter/Emora-Project#5]
Current Behavior:

Chrome prevents download of the file.
After download Windows Defender pops a message showing threats were found.
Virus total flags as virus also: https://www.virustotal.com/gui/file/19e352ee48e427abee64454eb7236941a8c7bcdf5db3aba853fa5604019755b8/detection

Additional Info
Operating System: Windows 10

**Comment by Prescoter** [comment]
Hey, thanks for reporting this
I think these detections are happening because of Inno Setup, the software I used to create the software setup, and also because the build isn't signed (If anyone is worried about this, you can run it in a VM, or just don't download the setup and only use the portable version)
If anyone knows of similar software to create a setup that has fewer false detections, feel free to give me the name here, it would help a lot!

**Comment by alistairwo** [comment]
I think you can just use visual studio? I might give it a crack myself when I have some time, but I know very little about C#

**Comment by JohnnyJayJay** [comment]
> I think you can just use visual studio? I might give it a crack myself when I have some time, but I know very little about C#

@alistairwo 

This repo is part of a scam/malware operation. It got its stars/forks from promising people on Discord porn in return (yes, really):

![](https://github.com/JohnnyJayJay/JohnnyJayJay.github.io/blob/main/content/img/discord-nsfw/github-guide.jpg?raw=true)
![](https://github.com/JohnnyJayJay/JohnnyJayJay.github.io/blob/main/content/img/discord-nsfw/star-and-fork.jpg?raw=true)

The Discord server is promoted using compromised accounts. I have no idea what role the "maintainer" of this repo plays in this, but I find it pretty safe to assume they're completely in on it given the activity on their account.

**Comment by Prescoter** [comment]
Hey,
First of all, I'd like to clarify that I have absolutely no involvement in this discord server, if it exists, because it's very easy to modify text on an html page
Secondly, I absolutely don't see what your message has to do with this issue?
If you think this project contains a virus, you're free to believe it, but at least check what you're saying before participating in issues. If you'd tested the software, you'd have seen that it wasn't a virus or a scam. (I personally find your message quite insulting)
Sincerely,
Have a nice day

**Comment by JohnnyJayJay** [comment]
You had a really creative idea for how to farm GitHub stars, I'll give you that. But you gotta work on keeping the façade up better, my man.

**Comment by alistairwo** [comment]
@JohnnyJayJay That's interesting, I actually stumbled across this project ironically through a cybersecurity content aggregator and a cybersecurity blogger. Looks like there is a whole discord community involved as well, possibly many more projects.

@Prescoter Perhaps when I have some free time i'll do a more thorough analysis, but there are so many red flags around this you need to get sorted, I'm trying to give benefit of the doubt but it just seems super dodgy.

To anyone reading, if you want to be safe just use [Sherlock](https://github.com/sherlock-project/sherlock), it does the exact same thing as this project except it's command line based.

**Comment by JohnnyJayJay** [comment]
>  I actually stumbled across this project ironically through a cybersecurity content aggregator and a cybersecurity blogger. Looks like there is a whole discord community involved as well, possibly many more projects.

Interesting. Can you share some links? Seems like the operation is casting a wider net, then. For reference, the Discord servers had around 700 members when I discovered it.

> @Prescoter Perhaps when I have some free time i'll do a more thorough analysis, but there are so many red flags around this you need to get sorted, I'm trying to give benefit of the doubt but it just seems super dodgy.

To me, there is practically zero doubt left this person is fully aware of what is happening. Three things point towards that conclusion:
- Person claims to be a developer of 10 years, yet they created their GitHub account recently and have only created this project. No other references or links to other projects.
- Reaction to the accusation is *very* odd. If you were a serious person and someone showed you that hundreds of people are being pushed to star and fork your repo for shady purposes, would that not concern you at least a little bit? I certainly wouldn't go straight to "you just want to ruin the project's reputation" (what reputation?) like they did for other issues raising concerns (#2 )
- How likely is it that shady scammers decide to push a completely unrelated repository on GitHub?

The only other explanation would be that their account is somehow compromised (without their knowledge), but I don't really buy that.

**Comment by alistairwo** [comment]
@JohnnyJayJay Yes check out [here ](https://talkback.sh/resource/82761c0b-d225-4a16-88bc-a49211e414bc)- it then links to a median blog with more info.

Yeah very fair points

**Comment by Prescoter** [comment]
Hey,
First, regarding the original purpose of the issue, I have good news: Windows Defender no longer detects the setup as a virus.
![image](https://github.com/Prescoter/Emora-Project/assets/138883341/12a021c9-f054-4798-8781-f73c1efc787c)

Secondly, regarding your malware accusations, I understand that antivirus detections can panic you, but false positives are common...
You can check the code or decompile the executables (they're not obfuscated, of course, so you can't have any doubts about their reliability), you won't find anything malicous.
If you see malicious code, at least tell me where...

And I remind you once again that I had nothing to do with the server you sent, I didn't even know it, even if I agree that it may explain the shady stars, but I have NO way of acting on it, github doesn't allow you to delete stars.

**Comment by JohnnyJayJay** [comment]
Stop lying and trying to cover your ass. It's ridiculous. Yeah right: a bunch of scammers decided to randomly promote your repo to do what exactly? Annoy you? Ruin your credibility? Again, who exactly are you? You just decided on a whim to create a GitHub account and a repo stating you need 150 stars before you publish anything?

Look, I don't know what your intention is exactly. But I don't care, it is astonishingly obvious that you're in on this thing. No serious open source dev does what you did.

But here's an idea for how you can make yourself at least a little bit more believable: delete all binaries from the releases. Easy!

**Comment by Prescoter** [comment]
"No serious open source dev does what you did." -> I don't see what I've done different from the others but Ok, you're free to think what you want. If the purpose of this discord server was to ruin this github repo, they've succeeded well in any case.

"delete all binaries from the releases" -> No ? The purpose of this software is to be simpler than the others. Wouldn't it make sense to do that? 
Wouldn't it be better if you checked the binaries yourself? The code is absolutely not obfuscated and doesn't block any virtual machine or sandbox, so you can run as many tests as you like, just until you finally understand that they don't contain any malware.

**Comment by felix068** [comment]
I just stumbled upon this GitHub repo because I needed software like this. And I saw this issue, and since this project is in C# (a language I know well), I thought I'd take a look. 

After reading and following the conversation, yes, I think this GitHub account could be considered "suspicious," but creating an account for a good GitHub project doesn't seem entirely absurd either. So, since I'm a C# dev, I decided once and for all to analyze all the code in this project to see. 

To my surprise, there is very little code, so it didn't take me long. And to make sure there's no trickery, I didn't just look at the code in Form1.cs but all the code files (.cs) in case the virus is in Form1.Designer.cs, for example (which would be sneaky but possible). And despite searching everywhere, I didn't find anything problematic in the code. I can now assert: "there is no virus in this program." I even compiled the program myself to see if the hash matched the hash of the program in Releases and the file installed by the setup. And the hash matches, so the binary files in the releases are indeed those created with the code you see in the repo.

Anyway, relying on Windows Defender's detection isn't very reliable (or any other antivirus) when you see that antiviruses detect each other as viruses to avoid competition. When you see that some antivirus software, during Windows updates, detects certain new system files as viruses, I'm not so sure about the reliability of antivirus programs. Moreover, if only the setup was detected (which is the case), and furthermore, if it's no longer detected as a virus, it just means there was a false positive like so many others... and also relying on VirusTotal with 5 detections by unknown antivirus programs is probably worthless.

**Comment by felix068** [comment]
I understand your concerns about the suspicious origins of this project and its promotion being linked to shady activities on Discord. However, after thoroughly analyzing the source code, I haven't found anything malicious. The code is legitimate and does not contain any viruses or malware.  

It may very well be that ill-intentioned people decided to promote this project without its knowledge and without its consent, with the aim of harming it or for other motivations that we are unaware of. But ultimately, what matters is the end result - the published source code.

We can speculate about the motivations and circumstances around the promotion all we want, but if the project works correctly and its code is clean, is it that important? An open source project should be judged on its actual content, not external rumors.

As long as the code has nothing malicious and the project fulfills its intended function, I think we don't care about the stories around it. What matters is having a functional and trustworthy software, and according to my in-depth analysis, this project meets those criteria. The rest is just speculation.

**Comment by felix068** [comment]
> Nobody ever claimed the code contains malware.
@JohnnyJayJay 

I would like to point out that at the start of the conversation you said: 
> This repo is part of a scam/malware operation. It got its stars/forks from promising people on Discord porn in return (yes, really):

[(https://github.com/Prescoter/Emora-Project/issues/5#issuecomment-2048543962)](https://github.com/Prescoter/Emora-Project/issues/5#issuecomment-2048543962)

**Comment by JohnnyJayJay** [comment]
> > Nobody ever claimed the code contains malware.
> @JohnnyJayJay 
> 
> I would like to point out that at the start of the conversation you said: 
> > This repo is part of a scam/malware operation. It got its stars/forks from promising people on Discord porn in return (yes, really):
> 
> [(https://github.com/Prescoter/Emora-Project/issues/5#issuecomment-2048543962)](https://github.com/Prescoter/Emora-Project/issues/5#issuecomment-2048543962)
> 

I'm not sure if you're intentionally being disingenuous here.

Source code does not have to equal distributed binaries. And even if the current release is (as *you* claim) harmless, I don't think it's an unfounded claim to view some of this as malicious (again, see #2).

I want to stress this again: this repo is being pushed by clearly malicious actors. You can choose to ignore this, but no person interested in security should. The author can claim they have nothing to do with the shady promotion. Why should I believe that? There is no sensible reason why random scammers would see a random repo by a random person that says "please give me stars" and think "oh yeah, from the kindness of my heart I'm going to tell people to push this". One of the following two **MUST** be true:

- The author of this repo is collaborating with shady people, perhaps paying them to push their repo (for whatever reason)
- The people behind the promotion (who *also* distribute malware) stand to gain something else from this (binary files in releases are a very convenient attack vector), either right now or in the future

You can choose to ignore this. But, let me put it this way: if you work in security, that would be a reason to fire you.

Especially after the xz-utils situation last week, which clearly showed us that a) source code can be fine but b) a malicious actor can use GitHub's release mechanism to introduce malware at any time in the future.

I don't know your motivations @felix068 , but to anyone else reading this: just stay away from this repo. Just use sherlock. It's available in trusted package managers, well-established and easy to use as well.

**Comment by felix068** [comment]
> Source code does not have to equal distributed binaries.

in this message I said I checked the hashes

[https://github.com/Prescoter/Emora-Project/issues/5#issuecomment-2050969203](https://github.com/Prescoter/Emora-Project/issues/5#issuecomment-2050969203)
> I even compiled the program myself to see if the hash matched the hash of the program in Releases and the file installed by the setup. And the hash matches, so the binary files in the releases are indeed those created with the code you see in the repo.

**Comment by JohnnyJayJay** [comment]
> @JohnnyJayJay 
> > Source code does not have to equal distributed binaries.
> 
> in this message I said I checked the hashes
> 
> [https://github.com/Prescoter/Emora-Project/issues/5#issuecomment-2050969203](https://github.com/Prescoter/Emora-Project/issues/5#issuecomment-2050969203)
> > I even compiled the program myself to see if the hash matched the hash of the program in Releases and the file installed by the setup. And the hash matches, so the binary files in the releases are indeed those created with the code you see in the repo.

Again: whether this exe *currently* contains malware or not is of little significance to me. I wouldn't trust any executable that isn't vetted in some way. 

What *is* significant is what I've extensively laid out in my last comment: malicious actors are pushing this repo and the maintainer is involved in *some* capacity. That should be enough for any reasonable person to categorically stay away from this whole thing.

I feel like you're ignoring this very clear line of reasoning. I don't know why you keep defending a bag of red flags, but honestly at this point I don't care. Do what you want; I've stated my case and hope people see reason.

**Comment by Kuraky** [comment]
> @JohnnyJayJay
> 
> > Source code does not have to equal distributed binaries.
> 
> in this message I said I checked the hashes
> 
> [#5 (comment)](https://github.com/Prescoter/Emora-Project/issues/5#issuecomment-2050969203)
> 
> > I even compiled the program myself to see if the hash matched the hash of the program in Releases and the file installed by the setup. And the hash matches, so the binary files in the releases are indeed those created with the code you see in the repo.

Care to explain how did you verify Emora.Setup.exe is safe since it's source code is not included in this repo?

**Comment by felix068** [comment]
> > @JohnnyJayJay
> > > Source code does not have to equal distributed binaries.
> > 
> > 
> > in this message I said I checked the hashes
> > [#5 (comment)](https://github.com/Prescoter/Emora-Project/issues/5#issuecomment-2050969203)
> > > I even compiled the program myself to see if the hash matched the hash of the program in Releases and the file installed by the setup. And the hash matches, so the binary files in the releases are indeed those created with the code you see in the repo.
> 
> Care to explain how did you verify Emora.Setup.exe is safe since it's source code is not included in this repo?

I said I checked the file installed by the setup not the setup.. nuance
I also plan to analyze the setup as well to see, but why put a virus in the setup and not the portable version?

**Comment by Kuraky** [comment]
Because the portable version is incredibly easy to check by comparing hashes like you just did. A better question is why bother making an installer for such a small program.

**Comment by felix068** [comment]
> Because the portable version is incredibly easy to check by comparing hashes like you just did. A better question is why bother making an installer for such a small program.

yes I agree that it is not very useful. And apart from having problems because there is a "virus" (as I say I don't see why put the virus in the setup and not the rest)

so @Prescoter can you remove the setup to avoid any problems? (he doesn't seem active at the moment I think we'll have to wait)

**Comment by DscTerminator** [comment]
> Nobody ever claimed the code contains malware. But here are the facts: This person, whom we know nothing about, created a repository with no content except "please give me 150 stars", then, _by chance_, people on Discord using hacked accounts to promote illegally obtained pornography and scams decide to push this repo, telling everyone to star and fork.
> 
> The "maintainer's" reaction is one of pure dismissal.
> 
> I don't mean this personally, but if you understand all this - if you understand that this repo that came out of the blue, created by a person nobody knows, distributing binaries in its releases is being artificially "grown" in terms of GitHub popularity - and you think all is fine and there is no problem with using it, you either have malicious intent as well or are **very** naive.
> 
> I can't believe I have to say this.

@JohnnyJayJay
I'd like to point out something that seems to have escaped all of you: the discord server encouraging people to put stars on this repo is not owned by the owner of this github repo. I'm familiar with this type of scam, and it's nothing new. The people behind this server own a github star service, and the people who buy them have no idea how they get there. I've lost their support discord server, but if I remember correctly, they were selling 30 stars for 10€, with a maximum quantity of 600 stars.
So @Prescoter really didn't seem to know what was really going on back there. On the other hand, I don't understand why he didn't just say so, maybe he was afraid his github account would get banned if someone reported a message in which he admitted to buying stars on github.
(I also enjoyed your article on fake NSFW servers on discord, and I'll join your server soon to give you more information on them)

@Prescoter 
The project, although simple, is still well done, and I think it's a shame for you that you thought you could buy hundreds of github stars without any fuss. I've analyzed the setup, and it seems clean, but it's true that it's not really useful to have put one in.

**Comment by JohnnyJayJay** [comment]
> I'd like to point out something that seems to have escaped all of you: the discord server encouraging people to put stars on this repo is not owned by the owner of this github repo.

I believe you're mistaken - @Kuraky and I have found pretty convincing evidence that the maintainer of this repo is involved with both the Discord server and malware distribution. I'll follow this up with more info.

**Comment by Prescoter** [comment]
> I understand your concerns about the suspicious origins of this project and its promotion being linked to shady activities on Discord. However, after thoroughly analyzing the source code, I haven't found anything malicious. The code is legitimate and does not contain any viruses or malware.
> 
> It may very well be that ill-intentioned people decided to promote this project without its knowledge and without its consent, with the aim of harming it or for other motivations that we are unaware of. But ultimately, what matters is the end result - the published source code.
> 
> We can speculate about the motivations and circumstances around the promotion all we want, but if the project works correctly and its code is clean, is it that important? An open source project should be judged on its actual content, not external rumors.
> 
> As long as the code has nothing malicious and the project fulfills its intended function, I think we don't care about the stories around it. What matters is having a functional and trustworthy software, and according to my in-depth analysis, this project meets those criteria. The rest is just speculation.

I totally agree with you. The project itself, as well as these releases being perfectly clean, there's not even a fuss to be made.
By example, there are ads that make you think your PC is infected with a virus on several websites and then tell you to buy avast to remove them, and yet nobody's going to complain to avast and tell them their antivirus is a virus because they're pissed off at the way they advertise. Why would you do that here?



> Because the portable version is incredibly easy to check by comparing hashes like you just did. A better question is why bother making an installer for such a small program.

@Kuraky
Simply because some people prefer setups to portable versions, and the project aims to be accessible to everyone, in the simplest way possible.

And, as I've said many times before, instead of just saying "there might be a virus", please check that there is one, and if you find one, then come and complain here, but speculations like yours are absolutely useless.

**Comment by JohnnyJayJay** [comment]
You're involved in some shady dealings as laid out extensively in this thread, that much has been proven beyond a reasonable doubt. It's mind-boggingly audacious to try to spin this in your favour and pretend the accusations are baseless. We know you have something to do with the illegal porn Discord, we know you're using their scheme to farm stars. You are not fooling anyone here.

**Comment by dante1nferno** [comment]
> You're involved in some shady dealings as laid out extensively in this thread, that much has been proven beyond a reasonable doubt. It's mind-boggingly audacious to try to spin this in your favour and pretend the accusations are baseless. We know you have something to do with the illegal porn Discord, we know you're using their scheme to farm stars. You are not fooling anyone here.

thank you Johnny, now it's seems clear
